crates/orchestrator/src/tests/jobs/mod.rs

```diff
@@ -20,3 +25,103 @@ async fn create_job_fails_job_already_exists() {
 async fn create_job_fails_works_new_job() {
     // TODO
 }
+
+use std::str::FromStr;
+
+impl FromStr for JobStatus {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "Created" => Ok(JobStatus::Created),
+            "LockedForProcessing" => Ok(JobStatus::LockedForProcessing),
+            "PendingVerification" => Ok(JobStatus::PendingVerification),
+            "Completed" => Ok(JobStatus::Completed),
+            "VerificationTimeout" => Ok(JobStatus::VerificationTimeout),
+            "Failed" => Ok(JobStatus::Failed),
+            s if s.starts_with("VerificationFailed(") && s.ends_with(')') => {
+                let reason = s[19..s.len() - 1].to_string();
+                Ok(JobStatus::VerificationFailed(reason))
+            }
+            _ => Err(format!("Invalid job status: {}", s)),
+        }
+    }
+}
+
+impl FromStr for JobType {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "SnosRun" => Ok(JobType::SnosRun),
+            "DataSubmission" => Ok(JobType::DataSubmission),
+            "ProofCreation" => Ok(JobType::ProofCreation),
+            "ProofRegistration" => Ok(JobType::ProofRegistration),
+            "StateTransition" => Ok(JobType::StateTransition),
+            _ => Err(format!("Invalid job type: {}", s)),
+        }
+    }
+}
+
+#[rstest]
+#[case("DataSubmission", "Completed")] // code should panic here, how can completed move to dl queue ?
+#[case("SnosRun", "PendingVerification")]
+#[case("ProofCreation", "LockedForProcessing")]
+#[case("ProofRegistration", "Created")]
+#[case("DataSubmission", "Failed")]
+#[case("StateTransition", "Completed")]
+#[case("ProofCreation", "VerificationTimeout")]
+#[case("DataSubmission", "VerificationFailed()")]
+#[tokio::test]
+async fn test_handle_job_failure(#[case] job_type: JobType, #[case] job_status: JobStatus) -> color_eyre::Result<()> {
+    use color_eyre::eyre::eyre;
+
+    TestConfigBuilder::new().build().await;
+    dotenvy::from_filename("../.env.test")?;
```

Comment:
doesn't this happen in the builder already?

Reply:
Valid, removed.